mesonbuild/cmake/interpreter.py

```diff
@@ -469,35 +469,21 @@ def handle_frameworks(flags: T.List[str]) -> T.List[str]:
             if dep_tgt:
                 self.depends.append(dep_tgt)
 
-    def process_object_libs(self, obj_target_list: T.List['ConverterTarget'], linker_workaround: bool) -> None:
-        # Try to detect the object library(s) from the generated input sources
-        temp = [x for x in self.generated if any(x.name.endswith('.' + y) for y in obj_suffixes)]
-        stem = [x.stem for x in temp]
-        exts = self._all_source_suffixes()
-        # Temp now stores the source filenames of the object files
-        for i in obj_target_list:
-            source_files = [x.name for x in i.sources + i.generated]
-            for j in stem:
-                # On some platforms (specifically looking at you Windows with vs20xy backend) CMake does
-                # not produce object files with the format `foo.cpp.obj`, instead it skipps the language
-                # suffix and just produces object files like `foo.obj`. Thus we have to do our best to
-                # undo this step and guess the correct language suffix of the object file. This is done
-                # by trying all language suffixes meson knows and checking if one of them fits.
-                candidates = [j]
-                if not any(j.endswith('.' + x) for x in exts):
-                    mlog.warning('Object files do not contain source file extensions, thus falling back to guessing them.', once=True)
-                    candidates += [f'{j}.{x}' for x in exts]
-                if any(x in source_files for x in candidates):
-                    if linker_workaround:
-                        self._append_objlib_sources(i)
-                    else:
-                        self.includes += i.includes
-                        self.includes = list(OrderedSet(self.includes))
-                        self.object_libs += [i]
-                    break
-
-        # Filter out object files from the sources
-        self.generated = [x for x in self.generated if not any(x.name.endswith('.' + y) for y in obj_suffixes)]
+    def process_object_libs(self, obj_target_list: T.List['ConverterTarget'], linker_workaround: bool, backend_name: str) -> None:
+        if backend_name.startswith('vs'):
+            # Even if --buildtype=Release is passed to meson, $(Configuration)
+            # needs to be replaced with "Debug", not "Release".
```

Comment:
Could you shed some light on why this is the case?

Reply:
Because "Debug" is the first configuration that cmake lists in codemodel*.json.

That being said, I rewrote that part to make it less hacky anyway.